client/properties.lua

```diff
@@ -1 +1,777 @@
+        not propertyType and {type = 'checkbox', label = 'Furnished? (Not For Garages!)', checked = true},
+    }, {allowCancel = true})
+    if not generalOptions then return end
+    if generalOptions[1] == '' then generalOptions[1] = nil end
+    if propertyType then
+        generalOptions[4] = propertyType == 'garage'
+        generalOptions[5] = propertyType == 'ipl'
+    end
+
+    local propertyOptions = {
+        {type = 'select', clearable = false, label = "Interior", options = createInteriorsList(generalOptions[4], generalOptions[5])}
+    }
+    if not generalOptions[4] then
+        propertyOptions[2] = {type = 'number', label = "Storage Volume", description = "Size of the storage (Kg)", default = 50, min = 1}
+        propertyOptions[3] = {type = 'number', label = "Storage Size", description = "Number of slots the Storage has", default = 10, min = 1}
+        if Config.Properties.useTaxes then
+            propertyOptions[4] = {type = 'multi-select', label = "Taxes", description = "Adds a tax if the property has the selected feature", options = getTaxesList()}
+        end
+    end
+
+    local propertyCreation = lib.inputDialog('Property Creator', propertyOptions, {allowCancel = true})
+    if not propertyCreation then return end
+    if not propertyCreation[1] then
+        QBCore.Functions.Notify('You need to select an interior!', 'error')
+        return
+    end
+    if not coords then
+        local coord, heading = GetEntityCoords(cache.ped), GetEntityHeading(cache.ped)
+        coords = {x = coord.x, y = coord.y, z = coord.z, w = heading}
+        coords = GetRoundedCoords(coords)
+    end
+
+    local inputResult = {
+        name = generalOptions[1] or GetStreetNameFromHashKey(GetStreetNameAtCoord(coords.x, coords.y, coords.z)),
+        price = generalOptions[2],
+        rent = generalOptions[3],
+        garage = generalOptions[4] or nil,
+        furnished = generalOptions[4] and nil or generalOptions[5],
+        interior = propertyCreation[1],
+        maxweight = propertyCreation[2] or nil,
+        slots = propertyCreation[3] or nil,
+        appliedtaxes = getAppliedTaxesList(propertyCreation[4]) or nil,
+        coords = {x = coords.x, y = coords.y, z = coords.z, w = coords.w},
+    }
+    createProperty(inputResult)
+end)
+
+RegisterNetEvent('qbx-property:client:enterIplProperty', function(interior, propertyId, isVisit, propertyOptions)
+    local coords = Config.IPLS[interior].coords
+    DoScreenFadeOut(1500)
+    Wait(250)
+
+    SetEntityCoords(cache.ped, coords.entrance.x, coords.entrance.y, coords.entrance.z, true, false, false, false)
+    SetEntityHeading(cache.ped, coords.entrance.w)
+    if propertyOptions then
+        setupIPL(GetInteriorAtCoords(GetEntityCoords(cache.ped)), Config.IPLS[interior].style, propertyOptions)
+    end
+    DoScreenFadeIn(1500)
+    CreatePropertyInteriorZones(coords, propertyId, isVisit)
+end)
```

Comment:
Consider extracting a function for this. That way I can read it as simple cause an effect. "When I enterIplProperty, I LoadAndTeleportToIpl()" Saves the reader from having to read the details if they don't need to.